tests/Unit/SearchEngine/Driver/Elasticsearch/ElasticSearchConfigTest.php

```diff
@@ -41,6 +41,13 @@ protected function setUp(): void
     {
         $this->options = $this->createMock(ServiceOptionsInterface::class);
         $this->sut = new ElasticSearchConfig($this->options);
+        putenv('ELASTICSEARCH_HOSTS=');
+        putenv('ELASTICSEARCH_USERNAME=');
+        putenv('ELASTICSEARCH_PASSWORD=');
+        putenv('ELASTICSEARCH_PREFIX=');
+        putenv('ELASTICSEARCH_CLOUD_ID=');
+        putenv('ELASTICSEARCH_API_KEY_ID=');
+        putenv('ELASTICSEARCH_API_KEY=');
```

Comment:
@tikhanovichA This should not be necessary. If the variables need to exist, even if we get it from config, that is an issue. Can we remove it from the test to make sure it reflects a real case, please? 

Thank you!

Reply:
with github actions yes it will work, but not with your local

Reply:
Ok, good point. Just notice that `putenv('VAR=')` sets it to empty, but `putenv('VAR')` actually removes it. I think you wanted the latest

Reply:
If this is a problem, maybe it would be better to allow passing the array of env vars to the constructor of `ElasticSearchConfig`, and fall back to `$_ENV` if that's not provided. That way tests would work without modifying the env. Something like this:

```php
public function __construct(ServiceOptionsInterface $serviceOptions, array config=[])
{
    $this->serviceOptions = $serviceOptions;
    $this->hosts = $config['hosts'] ?? explode(' ', $_ENV[self::ENV_OPTION_HOSTS])
    // ...
}

```

That way we allow to set them explicitly when instantiating the class in the tests.